cas-server-integration-ignite/src/main/java/org/jasig/cas/ticket/registry/IgniteTicketRegistry.java

```diff
@@ -0,0 +1,307 @@
+package org.jasig.cas.ticket.registry;
+
+import org.jasig.cas.authentication.principal.Service;
+import org.jasig.cas.ticket.registry.encrypt.AbstractCrypticTicketRegistry;
+import org.jasig.cas.ticket.ServiceTicket;
+import org.jasig.cas.ticket.Ticket;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.ignite.cache.CachePeekMode;
+import org.apache.ignite.cache.query.QueryCursor;
+import org.apache.ignite.cache.query.ScanQuery;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteState;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.lang.IgniteBiPredicate;
+import org.springframework.beans.BeanInstantiationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.HashSet;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import javax.cache.Cache.Entry;
+import javax.cache.Cache;
+import javax.cache.expiry.CreatedExpiryPolicy;
+import javax.cache.expiry.Duration;
+
+
+/**
+ * <p>
+ * <a href="https://ignite.apache.org">Ignite</a> based distributed ticket registry.
+ * </p>
+ *
+ * <p>
+ * Use distinct caches for ticket granting tickets (TGT) and service tickets (ST) for:
+ * <ul>
+ *   <li>Tuning : use cache level time to live with different values for TGT an ST.</li>
+ *   <li>Monitoring : follow separately the number of TGT and ST.</li>
+ * </ul>
+ *
+ * @author tduehr
+ * @since 4.3.0`
+ */
+@Component("igniteTicketRegistry")
+public final class IgniteTicketRegistry extends AbstractCrypticTicketRegistry implements DisposableBean {
+
+    /**
+     * Service ticket cache name.
+     */
+    public static final String SERVICES_CACHE = "serviceTicketsCache";
+
+    /**
+     * Ticket granting ticket cache name.
+     */
+    public static final String TICKETS_CACHE = "ticketGrantingTicketsCache";
+
+    @Autowired
+    @Qualifier("igniteConfiguration")
+    private IgniteConfiguration igniteConfiguration;
+
+    private IgniteCache<String, Ticket> serviceTicketsCache;
```

Comment:
I guess you could define more specific caches. For example: `private IgniteCache<String, ServiceTicket> serviceTicketsCache;`.